drivers/fuel_gauge/sy24561/sy24561.c

```diff
@@ -0,0 +1,346 @@
+		LOG_ERR("SOC alarm threshold %" PRIu16 " should be >= %" PRIu16, percent_threshold,
+			percent_threshold_min);
+		return -EINVAL;
+	}
+
+	/* This comes from datasheet at page 6 */
+	uint16_t const alarm_threshold_mask = 0b11111;
+	uint16_t const alarm_threshold = 32 - percent_threshold;
+
+	config &= ~alarm_threshold_mask;
+	config |= alarm_threshold;
+	LOG_DBG("new config register: 0x%x", config);
+
+	if (sy24561_write_reg(dev, SY24561_REG_CONFIG, config) != 0) {
+		LOG_ERR("Impossible to write config register");
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int sy24561_set_temperature(const struct device *dev, uint16_t temperature_dK /* 0.1K */)
+{
+	LOG_DBG("Setting temperature to %" PRIdK(u16), dKARGS(temperature_dK));
+
+	uint16_t config;
+	int const ret = sy24561_get_config(dev, &config);
+
+	if (ret != 0) {
+		LOG_ERR("Failed to read config: %d", ret);
+		return ret;
+	}
+
+	/* This comes from datasheet at page 5 */
+	uint16_t const temperature_dK_max = CELSIUS_TO_DECI_KELVIN(60);
+	uint16_t const temperature_dK_min = CELSIUS_TO_DECI_KELVIN(-20);
+
+	if (temperature_dK > temperature_dK_max) {
+		LOG_ERR("Temperature %" PRIdK(u16) " should be <= %" PRIdK(u16),
+			dKARGS(temperature_dK), dKARGS(temperature_dK_max));
+		return -EINVAL;
+	}
+
+	if (temperature_dK < temperature_dK_min) {
+		LOG_ERR("Temperature %" PRIdK(u16) " should be >= %" PRIdK(u16),
+			dKARGS(temperature_dK), dKARGS(temperature_dK_min));
+		return -EINVAL;
+	}
+
+	/* This comes from datasheet at page 5 and 6 */
+	uint16_t const temperature_mask = 0xff << 8;
+	uint16_t const temperature_reg_value = (40 + DECI_KELVIN_TO_CELSIUS(temperature_dK)) << 8;
+
+	config &= ~temperature_mask;
+	config |= temperature_reg_value;
+	LOG_DBG("new config register: 0x%x", config);
+
+	if (sy24561_write_reg(dev, SY24561_REG_CONFIG, config) != 0) {
+		LOG_ERR("Impossible to write config register");
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+int sy24561_get_prop(const struct device *dev, fuel_gauge_prop_t prop,
+		     union fuel_gauge_prop_val *val)
+{
+	switch (prop) {
+	case FUEL_GAUGE_VOLTAGE:
+		return sy24561_get_voltage(dev, &val->voltage);
+	case FUEL_GAUGE_RELATIVE_STATE_OF_CHARGE:
+		return sy24561_get_soc(dev, &val->relative_state_of_charge);
+	case FUEL_GAUGE_STATUS:
+		return sy24561_get_status(dev, &val->fg_status);
+	case FUEL_GAUGE_CURRENT_DIRECTION:
+		return sy24561_get_current_direction(dev, &val->current_direction);
+	default:
+		LOG_ERR("Property %d not supported", (int)prop);
+		return -ENOTSUP;
+	}
+	return 0;
+}
+
+static int sy24561_set_prop(const struct device *dev, fuel_gauge_prop_t prop,
+			    union fuel_gauge_prop_val val)
+{
+	int ret = 0;
+
+	switch (prop) {
+	case FUEL_GAUGE_SBS_REMAINING_CAPACITY_ALARM:
+		/* Threshold is set by percent, not capacity, but no such prop is available, so we
+		 * recycle this one which has the nearest meaning
+		 */
+		return sy24561_set_alarm_threshold(dev, val.sbs_remaining_capacity_alarm);
```

Comment:
Same here, wrong property value
```suggestion
		return sy24561_set_alarm_threshold(dev, val.state_of_charge_alarm);
```